### ISSUE TITLE
Players are getting stuck in spectator mode

### ISSUE THREAD

**Issue by ArionOwO** [CuzIm1Tigaaa/Spectator#38]
Mods are getting stuck in spectator mode.

How to recreate the issue:

First, don't be OP; just have all the spec permissions. Next, go into spec mode and spectate someone in a different dimension that you're currently not in and after that, If you use any form of teleportation from another plugin like /spawn, /warp, or /home, and they then try to /spec afterward, they will be teleported back to where they originally were when they first went into spec but it won't put their gamemode back into survival, they can try /spec or /unspec over and over again and nothing will bring them back into survival mode unless their gamemode is manually set back to survival by an OP or in console also they can't be turn back into survival mode unless they keep doing /spec until it says "you are no longer in spectator-mode" because if an OP tries to change their gamemode to survival while it says "you are now in spectator-mode" it won't work.

**Comment by RillSoji** [comment]
I am seeing this problem pop up as well. In addition to the above described behavior it will also yeet the players inventory if they're put back into survival manually. Here was the chain of events we duplicated. 

Player A in the overworld at their base
Player B is in the nether
PlayerA: /spec
PlayerA: /spec PlayerB
PlayerA: /home Mainbase
In console PlayerA seemed to return to their base but was teleported almost 20k blocks outside the world border. 
PlayerA: /home Mainbase
PlayerA made it back to their base this time. 
PlayerA: /spec
PlayerA is now stuck in spec and their inventory was wiped. 

Console Logs from that: 
https://pastebin.com/uYHfyFyc 

We are technically on velocity but all backend servers are paper and we are not trying to spec across the network. 
Velocity #436
Paper 1.21.1 #85

Advanced Portals
AntiPopup
Armor Stand Editor Reborn
Chunky
Command Sync
CoreProtect
DiscordSRV
Dynmap
HuskHomes
HuskSync
LuckPerms - Bukkit
Maintenance Addon
MinecraftAuth
mobdmz
PinIt
PlaceholderAPI
PremiumVanish
ProtocolLib
RoleTied
Rose Loot
SDS- SooperDooperScooper
Signed Velocity
SimplePlayerUtils
Spark - Paper
Spectator
STHU
TAB-Bridge
TabTPS
Teaks Tweaks
Vault
Wandering Trades
WorldEdit


I can get more tests done if needed.

**Comment by CuzIm1Tigaaa** [comment]
Thanks for the information!
spectating across the network should not be possible. 
About the inventory, I am currently working on this issue.
The other issues are known but complicated to resolve. Also I am not at home as I am doing an internship in another country at the moment. I will still try to fix as soon as possible.

**Comment by RillSoji** [comment]
Yes, we are not attempting to spectate across the network. The staff and players are aware and we don't try. They know we have some things that are per-server only like this.  No worries there! 

Thank you so much for your work on this plugin. It's a great option for our SMP servers and clients. Let me know if there's anything I can do to help you out.

**Comment by CuzIm1Tigaaa** [comment]
The plugin was a bit improved overall (from code side lol)
please let me know if this still persists

**Comment by CuzIm1Tigaaa** [comment]
As they was no further reply, I'll close this as fixed. If anything occurs again, feel free to open a new issue.
